### PR TITLE
Implement OpenAI July 2023 Updates

### PR DESCRIPTION
**Pull request by liushuangls**
**Describe the change**
Implement OpenAI July 2023 Updates

Issue: #425

### CODE REVIEW

File: edits.go [hunk 1]
```diff
@@ -30,7 +30,10 @@ type EditsResponse struct {
 	Choices []EditsChoice `json:"choices"`
 }
 
-// Perform an API call to the Edits endpoint.
+// Edits Perform an API call to the Edits endpoint.
+/* Deprecated: Users of the Edits API and its associated models (e.g., text-davinci-edit-001 or code-davinci-edit-001)
+will need to migrate to GPT-3.5 Turbo by January 4, 2024.
```
Comment (vvatanabe) [inline edits.go:35]: @liushuangls I thought a comment should note that CreateChatCompletion func (Chat Completions API) is an alternative.

> Users of the Edits API and its associated models (e.g., text-davinci-edit-001 or code-davinci-edit-001) will need to migrate to GPT-3.5 Turbo by January 4, 2024. The Edits API beta was an early exploratory API, meant to enable developers to return an edited version of the prompt based on instructions. We took the feedback from the Edits API into account when developing gpt-3.5-turbo and the Chat Completions API, which can now be used for the same purpose:

Refs: https://openai.com/blog/gpt-4-api-general-availability

Comment (liushuangls) [reply edits.go:35]: @vvatanabe I added it, thanks for your suggestion.

Comment (vvatanabe) [reply edits.go:35]: @liushuangls You are always quick to work and helpful!

**Review by vvatanabe — APPROVED** [review]

File: completion.go [hunk 2]
```diff
@@ -17,29 +17,46 @@ var (
...
+	GPT3TextDavinci001 = "text-davinci-001"
+	// Deprecated: Will be shut down on January 04, 2024. Use gpt-3.5-turbo-instruct instead.
 	GPT3DavinciInstructBeta = "davinci-instruct-beta"
-	GPT3Davinci             = "davinci"
-	GPT3CurieInstructBeta   = "curie-instruct-beta"
-	GPT3Curie               = "curie"
-	GPT3Ada                 = "ada"
-	GPT3Babbage             = "babbage"
+	// Deprecated: Will be shut down on January 04, 2024. Use davinci-002 instead.
+	GPT3Davinci    = "davinci"
```
Comment (sashabaranov) [inline completion.go:47]: Hey, thank you for the PR. I've re-read OpenAI post and I think we might not need to warn about these non-numerated models:

> Applications using the stable model names for base GPT-3 models (ada, babbage, curie, davinci) will automatically be upgraded to the new models listed above on January 4, 2024. The new models will also be accessible in the coming weeks for early testing by specifying the following model names in API calls: ada-002, babbage-002, curie-002, davinci-002.

https://openai.com/blog/gpt-4-api-general-availability

Comment (liushuangls) [reply completion.go]: @sashabaranov Fix it.

**Review by sashabaranov — APPROVED** [review]

**Comment by sashabaranov** [comment]
Thank you! 🙏🏻

**Pull request merged**